packages/packages/recently-edited/src/components/top-bar/posts-list.tsx

```diff
@@ -1,18 +1,32 @@
 import {
+	Divider,
 	MenuItem,
+	ListItemIcon,
 	ListSubheader,
 	Typography,
+	CircularProgress,
 } from '@elementor/ui';
 
 import { __ } from '@wordpress/i18n';
 import DocTypeChip, { Props } from '../top-bar/chip-doc-type';
 import { Post } from '../../hooks/use-recent-posts';
+import usePage from '../../hooks/use-page';
+import { PlusIcon } from '@elementor/icons';
+import { useEffect } from 'react';
 
 export type RecentPostsProps = {
 	recentPosts: Post[];
 };
 
 export default function PostsList( { recentPosts }: RecentPostsProps ) {
+	const { createPage, pageData, isLoading } = usePage();
+
+	useEffect( () => {
+		if ( pageData ) {
+			window.location.href = pageData.edit_url;
+		}
+	}, [ pageData ] );
```

Comment:
Feels weird to rely on effect to redirect, where is the event?

Reply:
You can `createPage().then(() => // redirect )` or `usePage({ onCreated: () => // redirect })`